### DISCUSSION TITLE
About fh format

### DISCUSSION THREAD

**Discussion by sniafas** [alan-turing-institute/sktime#4471] [Q&A] [answered]
Hi everyone,
I would like to ask about something that confused me, about the proper _futurehorizon (fh)_ data format when performing .predict.

The docstring clearly suggests for **int, list, np.array or ForecastingHorizon**.
What is the intuition behind this implementation to support multiple data formats. 

Bonus: what an exogeneous data refers to and how it contributes in forecasting estimation

**Comment by fkiraly** [comment, marked as answer]
> What is the intuition behind this implementation to support multiple data formats.

from a design perspective, the `fh` should be `ForecastingHorizon`, it contains the full information to specify where you want to forecast.

`ForecastingHorizon` coerces iterables such a `list`, `np.array`, so allowing the users to input the latter makes specification statements easier to write and easier to read.

Further, `int` is a bracket-less version of a single-element list, this is a convenience choice.

> what an exogeneous data refers to and how it contributes in forecasting estimation

this is synonymous with "external regressors" (a term we avoid due to possibility of confusion with supervised regression) or "predictor variables"

for an explanation, see Hyndman's nice book which uses that last terminology
https://otexts.com/fpp3/data-methods.html
